services/libs/integrations/src/integrations/github/processStream.ts

```diff
@@ -0,0 +1,844 @@
+    reviewThreadId,
+    ctx.integration.token,
+  )
+
+  const result = await pullRequestReviewThreadCommentsQuery.getSinglePage(data.page)
+
+  // filter out activities without authors (such as bots)
+  result.data = result.data.filter((i) => (i as any).author?.login)
+
+  // handle next page
+  await handleNextPageStream(ctx, result)
+
+  // get additional member info for each comment
+  for (const record of result.data) {
+    const member = await prepareMember(record.author, ctx)
+
+    // publish data
+    await ctx.publishData<GithubApiData>({
+      type: GithubActivityType.PULL_REQUEST_REVIEW_THREAD_COMMENT,
+      data: record,
+      member,
+      repo: data.repo,
+    })
+  }
+}
+
+export const processPullCommitsStream: ProcessStreamHandler = async (ctx) => {
+  let result: GraphQlQueryResponse
+
+  const data = ctx.stream.data as GithubBasicStream
+  const pullRequestNumber = data.prNumber
+  const pullRequestCommitsQuery = new PullRequestCommitsQuery(
+    data.repo,
+    pullRequestNumber,
+    ctx.integration.token,
+  )
+
+  try {
+    result = await pullRequestCommitsQuery.getSinglePage(data.page)
+  } catch (err) {
+    ctx.log.warn(
+      {
+        err,
+        repo: data.repo,
+        pullRequestNumber,
+      },
+      'Error while fetching pull request commits. Trying again without additions.',
+    )
+    const pullRequestCommitsQueryNoAdditions = new PullRequestCommitsQueryNoAdditions(
+      data.repo,
+      pullRequestNumber,
+      ctx.integration.token,
+    )
+    result = await pullRequestCommitsQueryNoAdditions.getSinglePage(data.page)
+  }
+
+  // handle next page
+  await handleNextPageStream(ctx, result)
+
+  // getting additional member info for each commit
+  {
+    const response = result.data[0] as PullRequestCommit | PullRequestCommitNoAdditions
+    const commits = response.repository.pullRequest.commits.nodes
+
+    for (const record of commits) {
+      for (const author of record.commit.authors.nodes) {
+        if (!author || !author?.user || !author?.user?.login) {
+          // eslint-disable-next-line no-continue
+          continue
+        }
+        const member = await prepareMember(author.user, ctx)
+
+        // publish data
+        await ctx.publishData<GithubApiData>({
+          type: GithubActivityType.AUTHORED_COMMIT,
+          data: record,
+          sourceParentId: response.repository.pullRequest.id,
+          member,
+          repo: data.repo,
+        })
+      }
+    }
+  }
+}
+
+const processIssuesStream: ProcessStreamHandler = async (ctx) => {
+  const data = ctx.stream.data as GithubBasicStream
+  const issuesQuery = new IssuesQuery(data.repo, ctx.integration.token)
+  const result = await issuesQuery.getSinglePage(data.page)
+
+  // filter out activities without authors (such as bots)
+  result.data = result.data.filter((i) => (i as any).author?.login)
+
+  // handle next page
+  await handleNextPageStream(ctx, result)
+
+  for (const record1 of result.data) {
```

Comment:
NIT: let's give better names to `record1` and `record2`

Reply:
done